### PR TITLE
Refactor ceph_monitoring.py

### PR DESCRIPTION
**Pull request by git-harry**
The existing ceph_monitoring.py script uses the command 'ceph report' to
pull statistics from the Ceph cluster. This command can be relatively
resource intensive due to the volume of data returned [1] and so this
commit refactors the script to use several ceph sub-commands to pull the
relevant pieces of information.

[1] http://tracker.ceph.com/issues/11904

Closes-Issue: https://github.com/rcbops/rpc-openstack/issues/641

### CODE REVIEW

**Comment by andymcc** [comment]
Seems sensible and I've checked that the metrics returned by the refactor are the same as before so LGTM.

**Comment by git-harry** [comment]
@prometheanfire why did you close this?

**Comment by prometheanfire** [comment]
not sure...

**Comment by git-harry** [comment]
recheck_upgrade
check_maas

**Comment by git-harry** [comment]
recheck_ceph

**Comment by git-harry** [comment]
recheck_ceph

File: maas/plugins/ceph_monitoring.py [hunk 1]
```diff
@@ -70,47 +83,50 @@ def get_osd_statistics(report=None, osd_ids=None):
...
     # Collect num PGs and num healthy PGs
-    pgs = {'total': 0, 'active_clean': 0}
-    for pg in report['pgmap']['pg_stats']:
-        pgs['total'] += 1
-        if pg['state'] == 'active+clean':
-            pgs['active_clean'] += 1
+    pgs = {'total': ceph_status['pgmap']['num_pgs'], 'active_clean': 0}
+    for state in ceph_status['pgmap']['pgs_by_state']:
+        if state['state_name'] == 'active+clean':
+            pgs['active+clean'] = state['count']
```
Comment (mancdaz) [inline maas/plugins/ceph_monitoring.py]: should this be 
```pgs['active_clean'] = state['count']```

**Comment by mancdaz** [comment]
@git-harry looks ok, except for one metric - ```pgs_active_clean```.  See old vs new output:

old:
```
# ./ceph_monitoring-old.py --name client.raxmon --keyring /etc/ceph/ceph.client.raxmon.keyring cluster
status okay
metric cluster_health uint32 2
metric monmap_epoch uint32 1
metric osdmap_epoch uint32 16
metric osds_total uint32 3
metric osds_up uint32 3
metric osds_in uint32 3
metric osds_kb uint64 278277096
metric osds_kb_avail uint64 129545200
metric osds_kb_used uint64 134527124
metric pgs_active_clean uint32 512
metric pgs_total uint32 512
```

new:
```
# ./ceph_monitoring-new.py --name client.raxmon --keyring /etc/ceph/ceph.client.raxmon.keyring cluster
status okay
metric cluster_health uint32 2
metric monmap_epoch uint32 1
metric osdmap_epoch uint32 16
metric osds_total uint32 3
metric osds_up uint32 3
metric osds_in uint32 3
metric osds_kb_used uint64 134527124
metric osds_kb_avail uint64 129545200
metric osds_kb uint64 278277096
metric pgs_active_clean uint32 0
metric pgs_total uint32 512
metric pgs_active+clean uint32 512
```

Specifically:

old:
```
metric pgs_active_clean uint32 512
```

new
```
metric pgs_active_clean uint32 0
metric pgs_active+clean uint32 512
```

I've commented inline where I think the issue is

**Comment by mancdaz** [comment]
Once checks pass, I'm 👍 on this patchset:

```
# ./ceph_monitoring-new.py --name client.raxmon --keyring /etc/ceph/ceph.client.raxmon.keyring cluster
status okay
metric cluster_health uint32 2
metric monmap_epoch uint32 1
metric osdmap_epoch uint32 16
metric osds_total uint32 3
metric osds_up uint32 3
metric osds_in uint32 3
metric osds_kb_used uint64 134588240
metric osds_kb_avail uint64 129484084
metric osds_kb uint64 278277096
metric pgs_active_clean uint32 512
metric pgs_total uint32 512
```

**Pull request merged**